// @nrwl/angular/src/generators/storybook-configuration/lib/generate-storybook-configuration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateStorybookConfiguration = void 0;
const tslib_1 = require("tslib");
const devkit_1 = require("@nrwl/devkit");
const versions_1 = require("../../../utils/versions");
function generateStorybookConfiguration(tree, options) {
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        const { configurationGenerator } = (0, devkit_1.ensurePackage)('@nrwl/storybook', versions_1.nxVersion);
        return yield configurationGenerator(tree, {
            name: options.name,
            uiFramework: '@storybook/angular',
            configureCypress: options.configureCypress,
            linter: options.linter,
            cypressDirectory: options.cypressDirectory,
            tsConfiguration: options.tsConfiguration,
            configureTestRunner: options.configureTestRunner,
            storybook7Configuration: options.storybook7Configuration,
            configureStaticServe: options.configureStaticServe,
            skipFormat: true,
        });
    });
}
exports.generateStorybookConfiguration = generateStorybookConfiguration;
