// @nrwl/angular/src/utils/versions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.jasmineMarblesVersion = exports.typesNodeVersion = exports.jestPresetAngularVersion = exports.tsNodeVersion = exports.autoprefixerVersion = exports.postcssUrlVersion = exports.postcssPresetEnvVersion = exports.postcssImportVersion = exports.postcssVersion = exports.tailwindVersion = exports.angularEslintVersion = exports.moduleFederationNodeVersion = exports.typesExpressVersion = exports.expressVersion = exports.typesCorsVersion = exports.corsVersion = exports.ngUniversalVersion = exports.tsLibVersion = exports.angularJsVersion = exports.zoneJsVersion = exports.rxjsVersion = exports.ngrxVersion = exports.ngPackagrVersion = exports.angularDevkitVersion = exports.angularVersion = exports.nxVersion = void 0;
exports.nxVersion = require('../../package.json').version;
exports.angularVersion = '~15.2.0';
exports.angularDevkitVersion = '~15.2.0';
exports.ngPackagrVersion = '~15.2.2';
exports.ngrxVersion = '~15.3.0';
exports.rxjsVersion = '~7.8.0';
exports.zoneJsVersion = '~0.12.0';
exports.angularJsVersion = '1.7.9';
exports.tsLibVersion = '^2.3.0';
exports.ngUniversalVersion = '~15.1.0';
exports.corsVersion = '~2.8.5';
exports.typesCorsVersion = '~2.8.5';
exports.expressVersion = '~4.18.2';
exports.typesExpressVersion = '4.17.14';
exports.moduleFederationNodeVersion = '~0.10.1';
exports.angularEslintVersion = '~15.0.0';
exports.tailwindVersion = '^3.0.2';
exports.postcssVersion = '^8.4.5';
exports.postcssImportVersion = '~14.1.0';
exports.postcssPresetEnvVersion = '~7.5.0';
exports.postcssUrlVersion = '~10.1.3';
exports.autoprefixerVersion = '^10.4.0';
exports.tsNodeVersion = '10.9.1';
exports.jestPresetAngularVersion = '~13.0.0';
exports.typesNodeVersion = '16.11.7';
exports.jasmineMarblesVersion = '^0.9.2';
