// @nrwl/angular/src/generators/directive/directive.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.directiveGenerator = void 0;
const tslib_1 = require("tslib");
const devkit_1 = require("@nrwl/devkit");
const path_1 = require("../utils/path");
const utils_1 = require("../utils");
let tsModule;
function directiveGenerator(tree, schema) {
    var _a, _b, _c;
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        const projects = (0, devkit_1.getProjects)(tree);
        if (!projects.has(schema.project)) {
            throw new Error(`Project "${schema.project}" does not exist!`);
        }
        (0, path_1.checkPathUnderProjectRoot)(tree, schema.project, schema.path);
        const project = (0, devkit_1.readProjectConfiguration)(tree, schema.project);
        const path = (_a = schema.path) !== null && _a !== void 0 ? _a : `${project.sourceRoot}`;
        const directiveNames = (0, devkit_1.names)(schema.name);
        const selector = (_b = schema.selector) !== null && _b !== void 0 ? _b : buildSelector(tree, schema.name, (_c = schema.prefix) !== null && _c !== void 0 ? _c : project.prefix);
        const pathToGenerateFiles = schema.flat
            ? './files/__directiveFileName__'
            : './files';
        yield (0, devkit_1.generateFiles)(tree, (0, devkit_1.joinPathFragments)(__dirname, pathToGenerateFiles), path, {
            selector,
            directiveClassName: directiveNames.className,
            directiveFileName: directiveNames.fileName,
            standalone: schema.standalone,
            tpl: '',
        });
        if (schema.skipTests) {
            const pathToSpecFile = (0, devkit_1.joinPathFragments)(path, `${!schema.flat ? `${directiveNames.fileName}/` : ``}${directiveNames.fileName}.directive.spec.ts`);
            tree.delete(pathToSpecFile);
        }
        if (!schema.skipImport && !schema.standalone) {
            const modulePath = (0, utils_1.findModule)(tree, path, schema.module);
            (0, utils_1.addToNgModule)(tree, path, modulePath, directiveNames.fileName, `${directiveNames.className}Directive`, `${directiveNames.fileName}.directive`, 'declarations', schema.flat, schema.export);
        }
        if (!schema.skipFormat) {
            yield (0, devkit_1.formatFiles)(tree);
        }
    });
}
exports.directiveGenerator = directiveGenerator;
function buildSelector(tree, name, prefix) {
    let selector = (0, devkit_1.names)(name).fileName;
    const selectorPrefix = (0, devkit_1.names)(prefix !== null && prefix !== void 0 ? prefix : (0, devkit_1.readNxJson)(tree).npmScope).fileName;
    return (0, devkit_1.names)(`${selectorPrefix}-${selector}`).propertyName;
}
exports.default = directiveGenerator;
