// @nrwl/angular/src/generators/pipe/pipe.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.pipeGenerator = void 0;
const tslib_1 = require("tslib");
const devkit_1 = require("@nrwl/devkit");
const path_1 = require("../utils/path");
const utils_1 = require("../utils");
let tsModule;
function pipeGenerator(tree, schema) {
    var _a;
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        const projects = (0, devkit_1.getProjects)(tree);
        if (!projects.has(schema.project)) {
            throw new Error(`Project "${schema.project}" does not exist!`);
        }
        (0, path_1.checkPathUnderProjectRoot)(tree, schema.project, schema.path);
        const project = (0, devkit_1.readProjectConfiguration)(tree, schema.project);
        const path = (_a = schema.path) !== null && _a !== void 0 ? _a : `${project.sourceRoot}`;
        const pipeNames = (0, devkit_1.names)(schema.name);
        const pathToGenerateFiles = schema.flat
            ? './files/__pipeFileName__'
            : './files';
        yield (0, devkit_1.generateFiles)(tree, (0, devkit_1.joinPathFragments)(__dirname, pathToGenerateFiles), path, {
            pipeClassName: pipeNames.className,
            pipeFileName: pipeNames.fileName,
            pipePropertyName: pipeNames.propertyName,
            standalone: schema.standalone,
            tpl: '',
        });
        if (schema.skipTests) {
            const pathToSpecFile = (0, devkit_1.joinPathFragments)(path, `${!schema.flat ? `${pipeNames.fileName}/` : ``}${pipeNames.fileName}.pipe.spec.ts`);
            tree.delete(pathToSpecFile);
        }
        if (!schema.skipImport && !schema.standalone) {
            const modulePath = (0, utils_1.findModule)(tree, path, schema.module);
            (0, utils_1.addToNgModule)(tree, path, modulePath, pipeNames.fileName, `${pipeNames.className}Pipe`, `${pipeNames.fileName}.pipe`, 'declarations', schema.flat, schema.export);
        }
        if (!schema.skipFormat) {
            yield (0, devkit_1.formatFiles)(tree);
        }
    });
}
exports.pipeGenerator = pipeGenerator;
exports.default = pipeGenerator;
