// @nrwl/angular/src/generators/library/lib/add-project.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addProject = void 0;
const devkit_1 = require("@nrwl/devkit");
function addProject(tree, libraryOptions) {
    const project = {
        name: libraryOptions.name,
        root: libraryOptions.projectRoot,
        sourceRoot: (0, devkit_1.joinPathFragments)(libraryOptions.projectRoot, 'src'),
        prefix: libraryOptions.prefix,
        tags: libraryOptions.parsedTags,
        projectType: 'library',
        targets: {
            build: libraryOptions.buildable || libraryOptions.publishable
                ? {
                    executor: libraryOptions.publishable
                        ? '@nrwl/angular:package'
                        : '@nrwl/angular:ng-packagr-lite',
                    outputs: ['{workspaceRoot}/dist/{projectRoot}'],
                    options: {
                        project: `${libraryOptions.projectRoot}/ng-package.json`,
                    },
                    configurations: {
                        production: {
                            tsConfig: `${libraryOptions.projectRoot}/tsconfig.lib.prod.json`,
                        },
                        development: {
                            tsConfig: `${libraryOptions.projectRoot}/tsconfig.lib.json`,
                        },
                    },
                    defaultConfiguration: 'production',
                }
                : undefined,
        },
    };
    (0, devkit_1.addProjectConfiguration)(tree, libraryOptions.name, project);
    return project;
}
exports.addProject = addProject;
